datafusion/sql/src/planner.rs

```diff
@@ -3213,7 +3213,7 @@ mod tests {
             let sql = "SELECT CAST(10 AS DECIMAL(0))";
             let err = logical_plan(sql).expect_err("query should have failed");
             assert_eq!(
-                r##"Internal("Decimal(precision = 0, scale = 0) should satisty `0 < precision <= 38`, and `scale <= precision`.")"##,
+                r##"Internal("Decimal(precision = 0, scale = 0) should satisfy `-38 < precision <= 38`, and `scale <= precision`.")"##,
```

Comment:
I'am confused about the error message. 
The origin message is more reasonable for me, and the precision should be `precision>0 and precision<=38` in the datafusion.
cc @alamb @tustvold

Reply:
@HaoYang670 I remember we have seen this issue about `SELECT CAST(10 AS DECIMAL(0))` and use the error precision in the SQL

Reply:
The type of `precision` is `usize`, so it should be always `>=0`.

Reply:
this is caused by https://github.com/apache/arrow-datafusion/pull/4400/files#r1035496657